lib/screens/login/login_form.dart

```diff
@@ -59,70 +106,68 @@ class _LoginFormState extends State<LoginForm> {
           ),
         ),
         onPressed: _submitForm,
-        child: Text(_localizations.loginButton),
+        child: Consumer(
+          builder: (context, widgetRef, child) {
+            final loginViewModel = widgetRef.watch(loginViewModelProvider);
+            return (loginViewModel.isLoading && !loginViewModel.hasError) ||
+                    loginViewModel.isReloading
+                ? const CircularProgressIndicator()
+                : Text(_localizations.loginButton);
+          },
+        ),
       );
 
-  // TODO: Move the validation logic to the ViewModel
-  // upon implementing Integration task
-  String? _emailValidator(String? email) {
-    // Just use a simple rule, no need complex Regex!
-    if (email == null || email.isEmpty || !email.contains('@')) {
-      return _localizations.invalidEmailError;
-    }
-    return null;
-  }
-
-  String? _passwordValidator(String? password) {
-    if (password == null || password.isEmpty || password.length < 8) {
-      return _localizations.invalidPasswordError;
-    }
-    return null;
-  }
-
-  bool _isFormSubmitted = false;
-
   void _submitForm() {
     setState(() => _isFormSubmitted = true);
 
-    if (_formKey.currentState!.validate()) {
-      KeyboardManager.dismiss(context);
-      // TODO - implement login
-      showAlertDialog(
-          context: context,
-          title: _localizations.loginFailAlertTitle,
-          message: _localizations.loginFailAlertMessage,
-          actions: [
-            TextButton(
-              child: Text(_localizations.okText),
-              onPressed: () => Navigator.pop(context),
-            ),
-          ]);
+    final isReadyToLogin =
+        (ref.watch(isValidEmailStreamProvider).value ?? false) &&
+            (ref.watch(isValidPasswordStreamProvider).value ?? false);
+    KeyboardManager.dismiss(context);
+    if (isReadyToLogin) {
+      ref.read(loginViewModelProvider.notifier).login();
```

Comment:
Then we don't need the email or password stream variable here. Just check it by using `_formKey`:
```suggestion
    KeyboardManager.dismiss(context);
    if (_formKey.currentState!.validate()) {
      ref.read(loginViewModelProvider.notifier).login();
```